pkg/tf2pulumi/convert/tf.go

```diff
@@ -1994,6 +1989,90 @@ func (s *scopes) pulumiName(fullyQualifiedPath string) string {
 	return getInner(currentSchema, currentInfo, parts[1:])
 }
 
+// Given a fully typed path (e.g. data.simple_data_source.a_field) returns whether a_field has max items equals one
+func (s *scopes) maxItemsOne(fullyQualifiedPath string) bool {
+	parts := strings.Split(fullyQualifiedPath, ".")
+	contract.Assertf(len(parts) != 0, "empty path passed into pulumiName")
+	contract.Assertf(parts[0] != "", "empty path part passed into pulumiName")
+
+	var getInner func(sch shim.SchemaMap, info map[string]*tfbridge.SchemaInfo, parts []string) bool
+	getInner = func(sch shim.SchemaMap, info map[string]*tfbridge.SchemaInfo, parts []string) bool {
+		contract.Assertf(parts[0] != "", "empty path part passed into pulumiName")
+
+		// Lookup the info for this part
+		var curSch shim.Schema
+		if sch != nil {
+			curSch = sch.Get(parts[0])
+		}
+		curInfo := info[parts[0]]
+
+		// We want the name of this part
```

Comment:
I feel like this core recurse loop could be shared, the only part that differs is this central return part.